pilot/pkg/security/model/authentication.go

```diff
@@ -285,6 +287,33 @@ func ApplyToCommonTLSContext(tlsContext *tls.CommonTlsContext, proxy *model.Prox
 	}
 }
 
+// SANToExactMatchWithFallback creates prefix matchs as fallbacks in addition to the converted exact matches,
+// if SAN inputs are SPIFFE URIs and end with the valid wildcard.
+func SANToExactMatchWithFallback(in []string) []*matcher.StringMatcher {
+	if len(in) == 0 {
+		return nil
+	}
+
+	res := make([]*matcher.StringMatcher, 0, len(in))
+	for _, s := range in {
+		// Replace e.g. "foo" with a exact match
+		res = append(res, &matcher.StringMatcher{
+			MatchPattern: &matcher.StringMatcher_Exact{Exact: s},
+		})
+
+		// Replace e.g. "spiffe://foo.cluster.domain/*" with prefix match
```

Comment:
the API docs said `spiffe://foo.cl*` was not allowed but I don't see anywhere that validates that?

Reply:
`spiffe://foo.cl*` would anyway create an exact match but never a prefix match, this is what Istio does now (no check on SPIFFE URI SAN, and just applies) and the one which is called `allowed`. It is guaranteed by only create one prefix match when with `/*` suffix.